fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -810,18 +806,17 @@ public void disableVersioning() {
 
     }
 
+    public static Predicate<Node> isVersioned =
+            UncheckedPredicate.uncheck((final Node node) -> node.isNodeType("mix:versionable"));
+
     @Override
     public boolean isVersioned() {
-        try {
-            return node.isNodeType("mix:versionable");
-        } catch (final RepositoryException e) {
-            throw new RepositoryRuntimeException(e);
-        }
+        return isVersioned.test(node);
```

Comment:
I would leave this as the logic-bearing part:
```
public static boolean isVersioned(Node node) {
    try {
        return node.isNodeType("mix:versionable");
    } catch (final RepositoryException e) {
        throw new RepositoryRuntimeException(e);
    }
}

@Override
public boolean isVersioned() {
    return isVersioned(node);
}
```
and instead of introducing a new `Predicate<Node>`-valued member above, use a method reference `FedoraResourceImpl::isVersioned`. That would read much better to me.